src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_Canvas_Measurement.cs

```diff
@@ -0,0 +1,125 @@
+		}
+
+#if !__WASM__
+		[TestMethod]
+
+		public async void When_Verify_Canvas_With_Outer_Clip()
+		{
+			var canvas = new Canvas_With_Outer_Clip();
+			WindowHelper.WindowContent = canvas;
+			WindowHelper.WaitForLoaded(canvas);
+
+			var renderer = new RenderTargetBitmap();
+			await WindowHelper.WaitForIdle();
+			await renderer.RenderAsync(canvas);
+			var bitmap = await RawBitmap.From(renderer, canvas);
+
+			var clippedLocation = canvas.Get_LocatorBorder1();
+			await WindowHelper.WaitForIdle();
+			ImageAssert.HasColorAtChild(bitmap, clippedLocation, (float)clippedLocation.Width / 2, (float)clippedLocation.Height / 2, Red);
+
+			var unclippedLocation = canvas.Get_LocatorBorder2();
+			await WindowHelper.WaitForIdle();
+			ImageAssert.HasColorAtChild(bitmap, unclippedLocation, (float)unclippedLocation.Width / 2, (float)unclippedLocation.Height / 2, Blue);
+		}
+
+		[TestMethod]
+		public async void When_Verify_Canvas_ZIndex()
+		{
+			var canvas = new CanvasZIndex();
+			WindowHelper.WindowContent = canvas;
+			WindowHelper.WaitForLoaded(canvas);
+
+			var renderer = new RenderTargetBitmap();
+			await WindowHelper.WaitForIdle();
+			await renderer.RenderAsync(canvas);
+			var bitmap = await RawBitmap.From(renderer, canvas);
+
+			var redBorderRect1 = canvas.Get_CanvasBorderRed1();
+			ImageAssert.HasColorAtChild(bitmap, redBorderRect1, (float)redBorderRect1.Width / 2, (float)redBorderRect1.Height / 2, Green);
+
+			var redBorderRect2 = canvas.Get_CanvasBorderRed2();
+			ImageAssert.HasColorAtChild(bitmap, redBorderRect2, (float)redBorderRect2.Width / 2, (float)redBorderRect2.Height / 2, Green);
+
+#if !__ANDROID__ //Android doesn't support Canvas.ZIndex on any panel
+			var redBorderRect3 = canvas.Get_CanvasBorderRed3();
+			ImageAssert.HasColorAtChild(bitmap, redBorderRect3, (float)redBorderRect3.Width / 2, (float)redBorderRect3.Height / 2, Green);
+#endif
+
+			var greenBorderRect1 = canvas.Get_CanvasBorderGreen1();
+			ImageAssert.HasColorAtChild(bitmap, greenBorderRect1, (float)greenBorderRect1.Width / 2, (float)greenBorderRect1.Height / 2, Brown);
+			ImageAssert.HasColorAtChild(bitmap, greenBorderRect1, (float)greenBorderRect1.Width - 1, (float)greenBorderRect1.Height / 2, Blue);
+
+			var greenBorderRect2 = canvas.Get_CanvasBorderGreen1();
+			ImageAssert.HasColorAtChild(bitmap,greenBorderRect2, (float)greenBorderRect2.Width / 2, (float)greenBorderRect2.Height / 2, Brown);
+			ImageAssert.HasColorAtChild(bitmap,greenBorderRect2, (float)greenBorderRect2.Width - 1, greenBorderRect2.Height / 2, Blue);
+#if !__ANDROID__ //Android doesn't support Canvas.ZIndex on any panel
+			
+			var greenBorderRect3 = canvas.Get_CanvasBorderGreen3();
+			ImageAssert.HasColorAtChild(bitmap, greenBorderRect3, (float)greenBorderRect3.Width / 2, (float)greenBorderRect3.Height / 2, Brown);
+			ImageAssert.HasColorAtChild(bitmap, greenBorderRect3, (float)greenBorderRect3.Width - 1, (float)greenBorderRect3.Height / 2, Blue);
+#endif
+		}
```

Comment:
```suggestion
		}

```

Reply:
I have added the extra empty line but it did not outdated the issue. Thank you for your help.